### ISSUE TITLE
GPU performance in spleeter

### ISSUE THREAD

**Issue by Chitvan-baish** [deezer/spleeter#700]
Hi @romi1502
When i am running spleeter latest version

here is my debug result, when running spleeter library

INFO:tensorflow:Using config: {'_model_dir': 'pretrained_models/4stems', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_save_checkpoints_secs': 600, '_session_config': gpu_options {
per_process_gpu_memory_fraction: 0.7
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_checkpoint_save_graph_def': True, '_service': None, '_cluster_spec': ClusterSpec({}), '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_types is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_shapes is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Apply unet for vocals_spectrogram
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/tensorflow/python/keras/layers/normalization.py:534: _colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
INFO:tensorflow:Apply unet for drums_spectrogram
INFO:tensorflow:Apply unet for bass_spectrogram
INFO:tensorflow:Apply unet for other_spectrogram
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from pretrained_models/4stems/model
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
DEBUG:spleeter:Writing file output5/temp11/vocals.wav
DEBUG:spleeter:Writing file output5/temp11/drums.wav
DEBUG:spleeter:Writing file output5/temp11/bass.wav
DEBUG:spleeter:Writing file output5/temp11/other.wav
INFO:spleeter:File output5/temp11/vocals.wav written succesfully
INFO:spleeter:File output5/temp11/drums.wav written succesfully
INFO:spleeter:File output5/temp11/bass.wav written succesfully
INFO:spleeter:File output5/temp11/other.wav written succesfully
Total Time taken in seconds 13

is that, how is suppose to work or am I missing something because when i am seeing my gpu utilization goes upto 77%

referenced #695 

and also, has spleeter-gpu library deprecated ? #690

can you share benchmarking on running spleeter using gpu, If yes, Please provide here

**Comment by romi1502** [comment]
Hi @Chitvan-baish,

So first, you need to know that if you're using the CLI, each time you're running the command, it will load the model again and includes the overhead. If you want to avoid the overhead, either separate all the file you want to separate with a single call to the CLI tool or use the python API, create a single `Separator` object and call the `separate` or `separate_to_file` method as many times as you wish.
Second, yes, `spleeter-gpu` is deprecated.

Finally, I've just done this very simple benchmark on a GTX1080, in a ipython console:
``` 
In [1]: from spleeter.separator import Separator

In [2]: separator = Separator('spleeter:4stems')

In [3]: separator.separate_to_file('audio_5min.wav', '/tmp') #First separation is longer is it involves model loading/compilation

In [4]: %timeit separator.separate_to_file('audio_5min.wav', '/tmp')
2.73 s ± 42.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

As you can see, the 5 minute file is on average separated in ~2.7s.
